test/tracerEq/test_h-advection_mes_2d.py

```diff
@@ -0,0 +1,207 @@
+    options.use_nonlinear_equations = False
+    options.use_lax_friedrichs_velocity = True
+    options.lax_friedrichs_velocity_scaling_factor = Constant(1.0)
+    options.use_lax_friedrichs_tracer = False
+    options.horizontal_velocity_scale = Constant(abs(u))
+    options.no_exports = True
+    options.output_directory = outputdir
+    options.simulation_end_time = t_end
+    options.simulation_export_time = t_export
+    options.solve_tracer = True
+    options.use_limiter_for_tracers = True
+    options.fields_to_export = ['tracer_2d']
+    options.update(model_options)
+    uv_expr = as_vector((u, 0))
+    bnd_salt_2d = {'value': Constant(0.0), 'uv': uv_expr}
+    bnd_uv_2d = {'uv': uv_expr}
+    solverobj.bnd_functions['tracer'] = {
+        1: bnd_salt_2d,
+        2: bnd_salt_2d,
+    }
+    solverobj.bnd_functions['momentum'] = {
+        1: bnd_uv_2d,
+        2: bnd_uv_2d,
+    }
+
+    solverobj.create_equations()
+
+    t = t_init  # simulation time
+
+    x0 = 0.3*lx
+    sigma = 1600.
+    xy = SpatialCoordinate(solverobj.mesh2d)
+    t_const = Constant(t)
+    ana_tracer_expr = exp(-(xy[0] - x0 - u*t_const)**2/sigma**2)
+
+    tracer_ana = Function(solverobj.function_spaces.H_2d, name='tracer analytical')
+    tracer_ana_p1 = Function(solverobj.function_spaces.P1_2d, name='tracer analytical')
+
+    p1dg_ho = FunctionSpace(solverobj.mesh2d, 'DG', options.polynomial_degree + 2,
+                            vfamily='DG', vdegree=options.polynomial_degree + 2)
+    tracer_ana_ho = Function(p1dg_ho, name='tracer analytical')
+
+    uv_init = Function(solverobj.function_spaces.U_2d, name='initial uv')
+    uv_init.project(uv_expr)
+    solverobj.assign_initial_conditions(uv=uv_init, tracer=ana_tracer_expr)
+
+    # export analytical solution
+    if not options.no_exports:
+        out_tracer_ana = File(os.path.join(options.output_directory, 'tracer_ana.pvd'))
+
+    def export_func():
+        if not options.no_exports:
+            solverobj.export()
+            # update analytical solution to correct time
+            t_const.assign(t)
+            tracer_ana.project(ana_tracer_expr)
+            out_tracer_ana.write(tracer_ana_p1.project(tracer_ana))
+
+    # export initial conditions
+    export_func()
+
+    # custom time loop that solves tracer equation only
+    ti = solverobj.timestepper.timesteppers.tracer
+
+    i = 0
+    iexport = 1
+    next_export_t = t + solverobj.options.simulation_export_time
+    while t < t_end - 1e-8:
+        ti.advance(t)
+        t += solverobj.dt
+        i += 1
+        if t >= next_export_t - 1e-8:
+            print_output('{:3d} i={:5d} t={:8.2f} s salt={:8.2f}'.format(iexport, i, t, norm(solverobj.fields.tracer_2d)))
+            export_func()
+            next_export_t += solverobj.options.simulation_export_time
+            iexport += 1
+
+    # project analytical solultion on high order mesh
+    t_const.assign(t)
+    tracer_ana_ho.project(ana_tracer_expr)
```

Comment:
I think projection to high-order function space is no longer necessary if `tracer_expr` is pure UFL expression; `errornorm` should just work. This is a remnant of times when it didn't.